### PR TITLE
[SIG-2829] Optional label

### PR DESCRIPTION
**Pull request by janjaap**
This PR fixes an issue in the incident form where the 'optional' indicator would not be shown. Existing tests for the `Header` component have been replaced to remove snapshot tests.

### CODE REVIEW

File: src/signals/incident/components/form/Header/index.test.js [hunk 1]
```diff
@@ -1,93 +1,122 @@
...
+    expect(queryByText(`U kunt maximaal ${requiredLength} tekens invoeren.`)).not.toBeInTheDocument();
 
-      expect(wrapper).toMatchSnapshot();
-    });
+    rerender(withAppContext(<Header hasError={hasError} getError={getError} touched />));
 
-    it('should render email error', () => {
-      hasError.mockImplementation(type => type === 'email');
-      wrapper.setProps({ touched: true });
+    expect(queryByText(`U kunt maximaal ${requiredLength} tekens invoeren.`)).toBeInTheDocument();
```
Comment (boris-arkenaar) [inline src/signals/incident/components/form/Header/index.test.js:97]: You could define the error message string in a `const` before, so you're absolutely sure you use the same string in both `expect` statements.

File: src/signals/incident/components/form/Header/index.test.js [hunk 2]
```diff
@@ -1,93 +1,122 @@
...
+    const hasError = prop => prop === 'required';
+    const touched = false;
 
-    it('should render optional field message', () => {
-      wrapper.setProps({ options: { validators: [] } });
+    const { queryByText } = render(withAppContext(<Header hasError={hasError} touched={touched} />));
 
-      expect(wrapper).toMatchSnapshot();
-    });
+    expect(queryByText('Dit is een verplicht veld')).not.toBeInTheDocument();
```
Comment (boris-arkenaar) [inline src/signals/incident/components/form/Header/index.test.js:120]: Where do you test the positive version?

Comment (janjaap) [reply src/signals/incident/components/form/Header/index.test.js:120]: In the 'should render required error' test case

File: src/signals/incident/components/form/Header/index.test.js (hunk 1 above)
Comment (janjaap) [reply src/signals/incident/components/form/Header/index.test.js:97]: I thought about that, but it's only two occurrences 🤷‍♂️

Comment (boris-arkenaar) [reply src/signals/incident/components/form/Header/index.test.js:97]: I know, but if you make a typo in the negative (`not`) test case, you won't notice it while your test isn't testing properly. So, still valuable, even if it's only two occurrences.

File: src/signals/incident/components/form/Header/index.test.js (hunk 2 above)
Comment (boris-arkenaar) [reply src/signals/incident/components/form/Header/index.test.js:120]: In this case as well, put the string in a `const`, to make sure the negative test tests what it should.

File: src/signals/incident/components/form/Header/index.test.js (hunk 1 above)
Comment (janjaap) [reply src/signals/incident/components/form/Header/index.test.js]: Done

**Pull request merged**